{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import nltk \n",
    "from collections import Counter\n",
    "from nltk.tokenize import word_tokenize\n",
    "import seaborn as sns\n",
    "from nltk.tokenize import word_tokenize\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "from wordcloud import wordcloud\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>ID</th>\n",
       "      <th>label</th>\n",
       "      <th>tanggal</th>\n",
       "      <th>judul</th>\n",
       "      <th>narasi</th>\n",
       "      <th>nama file gambar</th>\n",
       "      <th>data_cleaned</th>\n",
       "      <th>data</th>\n",
       "      <th>data_clean</th>\n",
       "      <th>databersih</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>4137</td>\n",
       "      <td>4137</td>\n",
       "      <td>95132.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Ustaz Di Cianjur Dipersekusi dan Dibacok</td>\n",
       "      <td>Kalian terlalu sibuk posting tentang Pilpres, ...</td>\n",
       "      <td>95132.jpg</td>\n",
       "      <td>sibuk posting pilpres sibuk posting    preside...</td>\n",
       "      <td>sibuk posting pilpres sibuk posting    preside...</td>\n",
       "      <td>sibuk posting pilpres sibuk posting  2019  pre...</td>\n",
       "      <td>sibuk posting pilpres sibuk posting  2019  pre...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4138</td>\n",
       "      <td>4138</td>\n",
       "      <td>840603.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Website Surveyfadlizon.com Milik Waketum Gerindra</td>\n",
       "      <td>Kalau ada surveyfadlizon.com itu bukan punya ...</td>\n",
       "      <td>840603.jpg</td>\n",
       "      <td>surveyfadlizon com pusat hoax nasional wakil k...</td>\n",
       "      <td>surveyfadlizon com pusat hoax nasional wakil k...</td>\n",
       "      <td>surveyfadlizon com pusat hoax nasional wakil k...</td>\n",
       "      <td>surveyfadlizon com pusat hoax nasional wakil k...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4139</td>\n",
       "      <td>4139</td>\n",
       "      <td>316308.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>: BBKSDA Jatim Mengklarifikasi Bahwa Wisata Al...</td>\n",
       "      <td>Menyusul beredarnya surat edaran rencana penut...</td>\n",
       "      <td>316308.jpg</td>\n",
       "      <td>susul edar surat edar rencana tutup kawah ijen...</td>\n",
       "      <td>susul edar surat edar rencana tutup kawah ijen...</td>\n",
       "      <td>susul edar surat edar rencana tutup kawah ijen...</td>\n",
       "      <td>susul edar surat edar rencana tutup kawah ijen...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4140</td>\n",
       "      <td>4140</td>\n",
       "      <td>365467.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>DIRUMORKAN PUNYA ATURAN BARU PEMBATASAN WAKTU ...</td>\n",
       "      <td>Kepala Humas BPJS Kesehatan, M. Iqbal Anas Ma...</td>\n",
       "      <td>365467.jpg</td>\n",
       "      <td>kepala humas bpjs sehat iqbal anas maruf uji l...</td>\n",
       "      <td>kepala humas bpjs sehat iqbal anas maruf uji l...</td>\n",
       "      <td>kepala humas bpjs sehat iqbal anas maruf uji l...</td>\n",
       "      <td>kepala humas bpjs sehat iqbal anas maruf uji l...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4141</td>\n",
       "      <td>4141</td>\n",
       "      <td>332690.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>DAFTAR NAMA HONORER YANG LOLOS SELEKSI CPNS 2018</td>\n",
       "      <td>Akhirnya pak bu",
       " inilah yang kita tunggu sekia...</td>\n",
       "      <td>332690.jpg</td>\n",
       "      <td>sekian</td>\n",
       "      <td>sekian</td>\n",
       "      <td>sekian</td>\n",
       "      <td>sekian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4142</td>\n",
       "      <td>4142</td>\n",
       "      <td>496530.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Akun Twitter @LawanPolitikJKW adalah Milik Fer...</td>\n",
       "      <td>Beberapa hari terakhir ini telah beredar luas ...</td>\n",
       "      <td>496530.jpg</td>\n",
       "      <td>edar luas capture buah twitter foto nama sah u...</td>\n",
       "      <td>edar luas capture buah twitter foto nama sah u...</td>\n",
       "      <td>edar luas capture buah twitter foto nama sah u...</td>\n",
       "      <td>edar luas capture buah twitter foto nama sah u...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4143</td>\n",
       "      <td>4143</td>\n",
       "      <td>832401.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Pollycarpus Bantah Jadi Kader Partai Berkarya</td>\n",
       "      <td>Setelah lama dikabarkan telah menjadi kader Pa...</td>\n",
       "      <td>832401.jpg</td>\n",
       "      <td>kabar kader partai karya pollycarpus budihari ...</td>\n",
       "      <td>kabar kader partai karya pollycarpus budihari ...</td>\n",
       "      <td>kabar kader partai karya pollycarpus budihari ...</td>\n",
       "      <td>kabar kader partai karya pollycarpus budihari ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4144</td>\n",
       "      <td>4144</td>\n",
       "      <td>160657.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Omongan Bin Masak Ngak DIperhatikan</td>\n",
       "      <td>Omongan Bin Masak Ngak DIperhatikan.. Ngeri N...</td>\n",
       "      <td>160657.jpg</td>\n",
       "      <td>omong bin masak ngak perhati ngeri ngeri sedap...</td>\n",
       "      <td>omong bin masak ngak perhati ngeri ngeri sedap...</td>\n",
       "      <td>omong bin masak ngak perhati ngeri ngeri sedap...</td>\n",
       "      <td>omong bin masak ngak perhati ngeri ngeri sedap...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4145</td>\n",
       "      <td>4145</td>\n",
       "      <td>504715.0</td>\n",
       "      <td>1</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Beginilah Caranya China Menghancurkan Dunia</td>\n",
       "      <td>Disinformasi yang pernah beberapa kali diedark...</td>\n",
       "      <td>504715.jpg</td>\n",
       "      <td>disinformasi edar pil china turki</td>\n",
       "      <td>disinformasi edar pil china turki</td>\n",
       "      <td>disinformasi edar pil china turki</td>\n",
       "      <td>disinformasi edar pil china turki</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4146</td>\n",
       "      <td>4146</td>\n",
       "      <td>480136.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Bappenas Bantah Tudingan Sandiaga Uno yang Seb...</td>\n",
       "      <td>Itu sama sekali tidak benar, mereka (BPS) pun...</td>\n",
       "      <td>480136.jpg</td>\n",
       "      <td>bps undang independensi menteri rencana bangun...</td>\n",
       "      <td>bps undang independensi menteri rencana bangun...</td>\n",
       "      <td>bps undang independensi menteri rencana bangun...</td>\n",
       "      <td>bps undang independensi menteri rencana bangun...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4147</td>\n",
       "      <td>4147</td>\n",
       "      <td>693138.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>diteriakkan nama Prabowo</td>\n",
       "      <td>Kok yang diteriakkan nama Prabowo??!!!</td>\n",
       "      <td>693138.jpg</td>\n",
       "      <td>teriak nama prabowo</td>\n",
       "      <td>teriak nama prabowo</td>\n",
       "      <td>teriak nama prabowo</td>\n",
       "      <td>teriak nama prabowo</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4148</td>\n",
       "      <td>4148</td>\n",
       "      <td>979851.0</td>\n",
       "      <td>0</td>\n",
       "      <td>24/08/2018</td>\n",
       "      <td>Ketua INASGOC Bantah Indonesia Bermain Curang ...</td>\n",
       "      <td>Atas tudingan bahwa Indonesia bermain curang p...</td>\n",
       "      <td>979851.jpg</td>\n",
       "      <td>tuding indonesia main curang cabang pencak sil...</td>\n",
       "      <td>tuding indonesia main curang cabang pencak sil...</td>\n",
       "      <td>tuding indonesia main curang cabang pencak sil...</td>\n",
       "      <td>tuding indonesia main curang cabang pencak sil...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0        ID  label     tanggal  \\\n",
       "4137        4137   95132.0      1  24/08/2018   \n",
       "4138        4138  840603.0      1  24/08/2018   \n",
       "4139        4139  316308.0      0  24/08/2018   \n",
       "4140        4140  365467.0      0  24/08/2018   \n",
       "4141        4141  332690.0      1  24/08/2018   \n",
       "4142        4142  496530.0      1  24/08/2018   \n",
       "4143        4143  832401.0      0  24/08/2018   \n",
       "4144        4144  160657.0      1  24/08/2018   \n",
       "4145        4145  504715.0      1  24/08/2018   \n",
       "4146        4146  480136.0      0  24/08/2018   \n",
       "4147        4147  693138.0      0  24/08/2018   \n",
       "4148        4148  979851.0      0  24/08/2018   \n",
       "\n",
       "                                                  judul  \\\n",
       "4137           Ustaz Di Cianjur Dipersekusi dan Dibacok   \n",
       "4138  Website Surveyfadlizon.com Milik Waketum Gerindra   \n",
       "4139  : BBKSDA Jatim Mengklarifikasi Bahwa Wisata Al...   \n",
       "4140  DIRUMORKAN PUNYA ATURAN BARU PEMBATASAN WAKTU ...   \n",
       "4141   DAFTAR NAMA HONORER YANG LOLOS SELEKSI CPNS 2018   \n",
       "4142  Akun Twitter @LawanPolitikJKW adalah Milik Fer...   \n",
       "4143      Pollycarpus Bantah Jadi Kader Partai Berkarya   \n",
       "4144                Omongan Bin Masak Ngak DIperhatikan   \n",
       "4145        Beginilah Caranya China Menghancurkan Dunia   \n",
       "4146  Bappenas Bantah Tudingan Sandiaga Uno yang Seb...   \n",
       "4147                           diteriakkan nama Prabowo   \n",
       "4148  Ketua INASGOC Bantah Indonesia Bermain Curang ...   \n",
       "\n",
       "                                                 narasi nama file gambar  \\\n",
       "4137  Kalian terlalu sibuk posting tentang Pilpres, ...        95132.jpg   \n",
       "4138  Kalau ada surveyfadlizon.com itu bukan punya ...       840603.jpg   \n",
       "4139  Menyusul beredarnya surat edaran rencana penut...       316308.jpg   \n",
       "4140  Kepala Humas BPJS Kesehatan, M. Iqbal Anas Ma...       365467.jpg   \n",
       "4141  Akhirnya pak bu\n",
       " inilah yang kita tunggu sekia...       332690.jpg   \n",
       "4142  Beberapa hari terakhir ini telah beredar luas ...       496530.jpg   \n",
       "4143  Setelah lama dikabarkan telah menjadi kader Pa...       832401.jpg   \n",
       "4144  Omongan Bin Masak Ngak DIperhatikan.. Ngeri N...       160657.jpg   \n",
       "4145  Disinformasi yang pernah beberapa kali diedark...       504715.jpg   \n",
       "4146  Itu sama sekali tidak benar, mereka (BPS) pun...       480136.jpg   \n",
       "4147             Kok yang diteriakkan nama Prabowo??!!!       693138.jpg   \n",
       "4148  Atas tudingan bahwa Indonesia bermain curang p...       979851.jpg   \n",
       "\n",
       "                                           data_cleaned  \\\n",
       "4137  sibuk posting pilpres sibuk posting    preside...   \n",
       "4138  surveyfadlizon com pusat hoax nasional wakil k...   \n",
       "4139  susul edar surat edar rencana tutup kawah ijen...   \n",
       "4140  kepala humas bpjs sehat iqbal anas maruf uji l...   \n",
       "4141                                             sekian   \n",
       "4142  edar luas capture buah twitter foto nama sah u...   \n",
       "4143  kabar kader partai karya pollycarpus budihari ...   \n",
       "4144  omong bin masak ngak perhati ngeri ngeri sedap...   \n",
       "4145                  disinformasi edar pil china turki   \n",
       "4146  bps undang independensi menteri rencana bangun...   \n",
       "4147                                teriak nama prabowo   \n",
       "4148  tuding indonesia main curang cabang pencak sil...   \n",
       "\n",
       "                                                   data  \\\n",
       "4137  sibuk posting pilpres sibuk posting    preside...   \n",
       "4138  surveyfadlizon com pusat hoax nasional wakil k...   \n",
       "4139  susul edar surat edar rencana tutup kawah ijen...   \n",
       "4140  kepala humas bpjs sehat iqbal anas maruf uji l...   \n",
       "4141                                             sekian   \n",
       "4142  edar luas capture buah twitter foto nama sah u...   \n",
       "4143  kabar kader partai karya pollycarpus budihari ...   \n",
       "4144  omong bin masak ngak perhati ngeri ngeri sedap...   \n",
       "4145                  disinformasi edar pil china turki   \n",
       "4146  bps undang independensi menteri rencana bangun...   \n",
       "4147                                teriak nama prabowo   \n",
       "4148  tuding indonesia main curang cabang pencak sil...   \n",
       "\n",
       "                                             data_clean  \\\n",
       "4137  sibuk posting pilpres sibuk posting  2019  pre...   \n",
       "4138  surveyfadlizon com pusat hoax nasional wakil k...   \n",
       "4139  susul edar surat edar rencana tutup kawah ijen...   \n",
       "4140  kepala humas bpjs sehat iqbal anas maruf uji l...   \n",
       "4141                                             sekian   \n",
       "4142  edar luas capture buah twitter foto nama sah u...   \n",
       "4143  kabar kader partai karya pollycarpus budihari ...   \n",
       "4144  omong bin masak ngak perhati ngeri ngeri sedap...   \n",
       "4145                  disinformasi edar pil china turki   \n",
       "4146  bps undang independensi menteri rencana bangun...   \n",
       "4147                                teriak nama prabowo   \n",
       "4148  tuding indonesia main curang cabang pencak sil...   \n",
       "\n",
       "                                             databersih  \n",
       "4137  sibuk posting pilpres sibuk posting  2019  pre...  \n",
       "4138  surveyfadlizon com pusat hoax nasional wakil k...  \n",
       "4139  susul edar surat edar rencana tutup kawah ijen...  \n",
       "4140  kepala humas bpjs sehat iqbal anas maruf uji l...  \n",
       "4141                                             sekian  \n",
       "4142  edar luas capture buah twitter foto nama sah u...  \n",
       "4143  kabar kader partai karya pollycarpus budihari ...  \n",
       "4144  omong bin masak ngak perhati ngeri ngeri sedap...  \n",
       "4145                  disinformasi edar pil china turki  \n",
       "4146  bps undang independensi menteri rencana bangun...  \n",
       "4147                                teriak nama prabowo  \n",
       "4148  tuding indonesia main curang cabang pencak sil...  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df=pd.read_csv ('dataset.csv')\n",
    "df.tail(12)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unnamed: 0            int64\n",
       "ID                  float64\n",
       "label                 int64\n",
       "tanggal              object\n",
       "judul                object\n",
       "narasi               object\n",
       "nama file gambar     object\n",
       "data_cleaned         object\n",
       "data                 object\n",
       "data_clean           object\n",
       "databersih           object\n",
       "dtype: object"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:2: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n",
      "  \n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\ipykernel_launcher.py:4: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "df_pos = df[df['label']==0]['databersih']\n",
    "df_pos.to_csv('nonhoax.csv', encoding='utf8', index = False)\n",
    "df_neg = df[df['label']==1]['databersih']\n",
    "df_neg.to_csv('hoax.csv', encoding='utf8', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Proporsi label\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYsAAAEHCAYAAABfkmooAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAASxUlEQVR4nO3df6zd9X3f8ecrNpB2SYsZN8yx3RqlblfStSS5A9ZIW360YJBak4pMoLV4DMmZBlsjVdVINY0MytRuSVGSZUiucDBtGmolzeJFrMylaaOsS+A6cwFDEHeEhVt7+KYmJCwqm+l7f5zPVU7se+/nQu4595r7fEhH5/t9fz7fc97Hsu5L3x/ne1JVSJK0mFetdAOSpNXPsJAkdRkWkqQuw0KS1GVYSJK6DAtJUtf6Ub1wklcDnwfOau/zyaq6OcldwD8AnmtT/3FVHUoS4EPAFcC3W/3L7bV2Av+qzf/1qtq72Hufe+65tXXr1mX+RJL0ynbw4MGvV9XEfGMjCwvgBeAdVfV8kjOALyT5L23sV6vqkyfNvxzY1h4XA3cAFyc5B7gZmAQKOJhkf1U9u9Abb926lampqWX+OJL0ypbkfy00NrLDUDXwfFs9oz0W+wbgDuDutt0XgbOTbAQuAw5U1fEWEAeA7aPqW5J0qpGes0iyLskh4BiDP/hfakO3JXkoye1Jzmq1TcDTQ5vPtNpC9ZPfa1eSqSRTs7Ozy/5ZJGktG2lYVNWLVXUhsBm4KMlPAO8D/jbwd4FzgH/Zpme+l1ikfvJ77a6qyaqanJiY95CbJOllGsvVUFX1DeBPgO1VdbQdanoB+BhwUZs2A2wZ2mwzcGSRuiRpTEYWFkkmkpzdlr8P+BngK+08BO3qpyuBR9om+4FrM3AJ8FxVHQXuAy5NsiHJBuDSVpMkjckor4baCOxNso5BKO2rqs8m+eMkEwwOLx0C/mmbfy+Dy2anGVw6ex1AVR1PcivwYJt3S1UdH2HfkqST5JV4i/LJycny0llJemmSHKyqyfnG/Aa3JKnLsJAkdY3ynIWkEfnaLX9npVvQKvRD//rhkb22exaSpC7DQpLUZVhIkroMC0lSl2EhSeoyLCRJXYaFJKnLsJAkdRkWkqQuw0KS1GVYSJK6DAtJUpdhIUnqMiwkSV2GhSSpy7CQJHUZFpKkLsNCktRlWEiSukYWFkleneSBJH+e5HCSf9Pq5yf5UpInkvx+kjNb/ay2Pt3Gtw691vta/fEkl42qZ0nS/Ea5Z/EC8I6q+ingQmB7kkuA3wRur6ptwLPA9W3+9cCzVfUjwO1tHkkuAK4G3ghsB/5jknUj7FuSdJKRhUUNPN9Wz2iPAt4BfLLV9wJXtuUdbZ02/s4kafV7quqFqvoqMA1cNKq+JUmnGuk5iyTrkhwCjgEHgP8JfKOqTrQpM8CmtrwJeBqgjT8H/M3h+jzbDL/XriRTSaZmZ2dH8XEkac0aaVhU1YtVdSGwmcHewI/PN609Z4Gxheonv9fuqpqsqsmJiYmX27IkaR5juRqqqr4B/AlwCXB2kvVtaDNwpC3PAFsA2vgPAseH6/NsI0kag1FeDTWR5Oy2/H3AzwCPAZ8DrmrTdgKfacv72zpt/I+rqlr96na11PnANuCBUfUtSTrV+v6Ul20jsLddufQqYF9VfTbJo8A9SX4d+B/AnW3+ncDvJJlmsEdxNUBVHU6yD3gUOAHcUFUvjrBvSdJJRhYWVfUQ8KZ56k8yz9VMVfVXwLsXeK3bgNuWu0dJ0tL4DW5JUpdhIUnqMiwkSV2GhSSpy7CQJHUZFpKkLsNCktRlWEiSugwLSVKXYSFJ6jIsJEldhoUkqcuwkCR1GRaSpC7DQpLUZVhIkroMC0lSl2EhSeoyLCRJXYaFJKnLsJAkdY0sLJJsSfK5JI8lOZzkl1v9/Un+Ismh9rhiaJv3JZlO8niSy4bq21ttOslNo+pZkjS/9SN87RPAr1TVl5O8FjiY5EAbu72qPjA8OckFwNXAG4HXA3+U5Efb8EeBnwVmgAeT7K+qR0fYuyRpyMjCoqqOAkfb8reSPAZsWmSTHcA9VfUC8NUk08BFbWy6qp4ESHJPm2tYSNKYjOWcRZKtwJuAL7XSjUkeSrInyYZW2wQ8PbTZTKstVJckjcnIwyLJa4BPAe+tqm8CdwBvAC5ksOfxwbmp82xei9RPfp9dSaaSTM3Ozi5L75KkgZGGRZIzGATFx6vqDwCq6pmqerGq/hr4bb5zqGkG2DK0+WbgyCL171JVu6tqsqomJyYmlv/DSNIaNsqroQLcCTxWVb81VN84NO1dwCNteT9wdZKzkpwPbAMeAB4EtiU5P8mZDE6C7x9V35KkU43yaqi3Ar8EPJzkUKv9GnBNkgsZHEp6CngPQFUdTrKPwYnrE8ANVfUiQJIbgfuAdcCeqjo8wr4lSScZ5dVQX2D+8w33LrLNbcBt89TvXWw7SdJo+Q1uSVKXYSFJ6jIsJEldhoUkqcuwkCR1GRaSpC7DQpLUZVhIkroMC0lSl2EhSeoyLCRJXYaFJKnLsJAkdRkWkqQuw0KS1GVYSJK6DAtJUpdhIUnqMiwkSV2GhSSpy7CQJHUZFpKkrpGFRZItST6X5LEkh5P8cqufk+RAkifa84ZWT5IPJ5lO8lCSNw+91s42/4kkO0fVsyRpfqPcszgB/EpV/ThwCXBDkguAm4D7q2obcH9bB7gc2NYeu4A7YBAuwM3AxcBFwM1zASNJGo+RhUVVHa2qL7flbwGPAZuAHcDeNm0vcGVb3gHcXQNfBM5OshG4DDhQVcer6lngALB9VH1Lkk41lnMWSbYCbwK+BJxXVUdhECjA69q0TcDTQ5vNtNpC9ZPfY1eSqSRTs7Ozy/0RJGlNG3lYJHkN8CngvVX1zcWmzlOrRerfXajaXVWTVTU5MTHx8pqVJM1rpGGR5AwGQfHxqvqDVn6mHV6iPR9r9Rlgy9Dmm4Eji9QlSWOypLBIcv9SaieNB7gTeKyqfmtoaD8wd0XTTuAzQ/Vr21VRlwDPtcNU9wGXJtnQTmxf2mqSpDFZv9hgklcD3w+c2/5Qzx0S+gHg9Z3XfivwS8DDSQ612q8BvwHsS3I98DXg3W3sXuAKYBr4NnAdQFUdT3Ir8GCbd0tVHV/ax5MkLYdFwwJ4D/BeBsFwkO+ExTeBjy62YVV9gfnPNwC8c575BdywwGvtAfZ0epUkjciiYVFVHwI+lOSfV9VHxtSTJGmV6e1ZAFBVH0ny08DW4W2q6u4R9SVJWkWWFBZJfgd4A3AIeLGVCzAsJGkNWFJYAJPABe28giRpjVnq9yweAf7WKBuRJK1eS92zOBd4NMkDwAtzxar6+ZF0JUlaVZYaFu8fZROSpNVtqVdD/emoG5EkrV5LvRrqW3zn5n1nAmcA/6eqfmBUjUmSVo+l7lm8dng9yZUMfohIkrQGvKy7zlbVfwLescy9SJJWqaUehvqFodVXMfjehd+5kKQ1YqlXQ/3c0PIJ4CkGP4MqSVoDlnrO4rpRNyJJWr2W+uNHm5N8OsmxJM8k+VSSzaNuTpK0Oiz1BPfHGPyS3euBTcB/bjVJ0hqw1LCYqKqPVdWJ9rgLmBhhX5KkVWSpYfH1JL+YZF17/CLwl6NsTJK0eiw1LP4J8A+B/w0cBa6i/Ua2JOmVb6mXzt4K7KyqZwGSnAN8gEGISJJe4Za6Z/GTc0EBUFXHgTeNpiVJ0mqz1LB4VZINcyttz2KpeyWSpNPcUsPig8CfJbk1yS3AnwH/brENkuxp38t4ZKj2/iR/keRQe1wxNPa+JNNJHk9y2VB9e6tNJ7nppX08SdJyWOo3uO9OMsXg5oEBfqGqHu1sdhfwH4C7T6rfXlUfGC4kuQC4Gngjg+9y/FGSH23DHwV+FpgBHkyyfwnvLUlaRks+lNT+QC/5j3RVfT7J1iVO3wHcU1UvAF9NMs13boE+XVVPAiS5p801LCRpjF7WLcq/Rzcmeagdppo7D7IJeHpozkyrLVQ/RZJdSaaSTM3Ozo6ib0las8YdFncAbwAuZPB9jQ+2euaZW4vUTy1W7a6qyaqanJjwy+WStJzGekVTVT0zt5zkt4HPttUZYMvQ1M3Akba8UF2SNCZj3bNIsnFo9V3A3JVS+4Grk5yV5HxgG/AA8CCwLcn5Sc5kcBJ8/zh7liSNcM8iySeAtwHnJpkBbgbeluRCBoeSngLeA1BVh5PsY3Di+gRwQ1W92F7nRuA+YB2wp6oOj6pnSdL8RhYWVXXNPOU7F5l/G3DbPPV7gXuXsTVJ0ku0EldDSZJOM4aFJKnLsJAkdRkWkqQuw0KS1GVYSJK6DAtJUpdhIUnqMiwkSV2GhSSpy7CQJHUZFpKkLsNCktRlWEiSugwLSVKXYSFJ6jIsJEldhoUkqcuwkCR1GRaSpC7DQpLUNbKwSLInybEkjwzVzklyIMkT7XlDqyfJh5NMJ3koyZuHttnZ5j+RZOeo+pUkLWyUexZ3AdtPqt0E3F9V24D72zrA5cC29tgF3AGDcAFuBi4GLgJungsYSdL4jCwsqurzwPGTyjuAvW15L3DlUP3uGvgicHaSjcBlwIGqOl5VzwIHODWAJEkjNu5zFudV1VGA9vy6Vt8EPD00b6bVFqpLksZotZzgzjy1WqR+6gsku5JMJZmanZ1d1uYkaa0bd1g80w4v0Z6PtfoMsGVo3mbgyCL1U1TV7qqarKrJiYmJZW9cktaycYfFfmDuiqadwGeG6te2q6IuAZ5rh6nuAy5NsqGd2L601SRJY7R+VC+c5BPA24Bzk8wwuKrpN4B9Sa4Hvga8u02/F7gCmAa+DVwHUFXHk9wKPNjm3VJVJ580lySN2MjCoqquWWDonfPMLeCGBV5nD7BnGVuTJL1Eq+UEtyRpFTMsJEldhoUkqcuwkCR1GRaSpC7DQpLUZVhIkroMC0lSl2EhSeoa2Te4T3dv+dW7V7oFrUIH//21K92CtCLcs5AkdRkWkqQuw0KS1GVYSJK6DAtJUpdhIUnqMiwkSV2GhSSpy7CQJHUZFpKkLsNCktRlWEiSulYkLJI8leThJIeSTLXaOUkOJHmiPW9o9ST5cJLpJA8lefNK9CxJa9lK7lm8vaourKrJtn4TcH9VbQPub+sAlwPb2mMXcMfYO5WkNW41HYbaAexty3uBK4fqd9fAF4Gzk2xciQYlaa1aqbAo4L8mOZhkV6udV1VHAdrz61p9E/D00LYzrfZdkuxKMpVkanZ2doStS9Las1I/fvTWqjqS5HXAgSRfWWRu5qnVKYWq3cBugMnJyVPGJUkv34rsWVTVkfZ8DPg0cBHwzNzhpfZ8rE2fAbYMbb4ZODK+biVJYw+LJH8jyWvnloFLgUeA/cDONm0n8Jm2vB+4tl0VdQnw3NzhKknSeKzEYajzgE8nmXv/36uqP0zyILAvyfXA14B3t/n3AlcA08C3gevG37IkrW1jD4uqehL4qXnqfwm8c556ATeMoTVJ0gJW06WzkqRVyrCQJHUZFpKkLsNCktRlWEiSugwLSVKXYSFJ6jIsJEldhoUkqcuwkCR1GRaSpC7DQpLUZVhIkroMC0lSl2EhSeoyLCRJXYaFJKnLsJAkdRkWkqQuw0KS1GVYSJK6DAtJUtdpExZJtid5PMl0kptWuh9JWktOi7BIsg74KHA5cAFwTZILVrYrSVo7TouwAC4Cpqvqyar6v8A9wI4V7kmS1oz1K93AEm0Cnh5anwEuHp6QZBewq60+n+TxMfW2FpwLfH2lm1gN8oGdK92CTuX/zzk353t9hR9eaOB0CYv5/gXqu1aqdgO7x9PO2pJkqqomV7oPaT7+/xyP0+Uw1AywZWh9M3BkhXqRpDXndAmLB4FtSc5PciZwNbB/hXuSpDXjtDgMVVUnktwI3AesA/ZU1eEVbmst8fCeVjP/f45Bqqo/S5K0pp0uh6EkSSvIsJAkdRkWWpS3WdFqlGRPkmNJHlnpXtYKw0IL8jYrWsXuAravdBNriWGhxXibFa1KVfV54PhK97GWGBZazHy3Wdm0Qr1IWkGGhRbTvc2KpLXBsNBivM2KJMCw0OK8zYokwLDQIqrqBDB3m5XHgH3eZkWrQZJPAP8d+LEkM0muX+meXum83Yckqcs9C0lSl2EhSeoyLCRJXYaFJKnLsJAkdRkW0jJI8nxnfOtLvUNqkruSXPW9dSYtD8NCktRlWEjLKMlrktyf5MtJHk4yfJfe9Un2JnkoySeTfH/b5i1J/jTJwST3Jdm4Qu1LCzIspOX1V8C7qurNwNuBDyaZuyHjjwG7q+ongW8C/yzJGcBHgKuq6i3AHuC2FehbWtT6lW5AeoUJ8G+T/H3grxnc0v28NvZ0Vf23tvy7wL8A/hD4CeBAy5R1wNGxdiwtgWEhLa9/BEwAb6mq/5fkKeDVbezke+sUg3A5XFV/b3wtSi+dh6Gk5fWDwLEWFG8Hfnho7IeSzIXCNcAXgMeBibl6kjOSvHGsHUtLYFhIy+vjwGSSKQZ7GV8ZGnsM2JnkIeAc4I72c7VXAb+Z5M+BQ8BPj7lnqcu7zkqSutyzkCR1GRaSpC7DQpLUZVhIkroMC0lSl2EhSeoyLCRJXf8f6JMIkWfle4kAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(\"Proporsi label\")\n",
    "ax = sns.countplot(df['label'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    3380\n",
       "0     769\n",
       "Name: label, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['label'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Bentuk VSM-nya\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "\n",
    "tfidf_vectorizer = TfidfVectorizer(lowercase=True, smooth_idf= True, sublinear_tf=True, \n",
    "                                   ngram_range=(1, 2), max_df=0.90, min_df=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4149, 12495) 4149\n"
     ]
    }
   ],
   "source": [
    "##TFIDF\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer\n",
    "import gensim\n",
    "#vectorizer = TfidfVectorizer(ngram_range=(1, 2))\n",
    "vectorizer = TfidfVectorizer(max_df=0.1, min_df=5, ngram_range=(1, 2))\n",
    "\n",
    "listdata=df['data_cleaned'].values.astype('object')\n",
    "listdata = [d for d in listdata]\n",
    "\n",
    "listdata\n",
    "v = TfidfVectorizer(decode_error='replace', encoding='utf-8')\n",
    "tfidf = v.fit_transform(df['data_cleaned'].values.astype('U'))\n",
    "y = df.iloc[:, 2].values\n",
    "print(tfidf.shape, len(y)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SMOTE(sampling_strategy='all')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#balancing\n",
    "from imblearn.over_sampling import RandomOverSampler\n",
    "from imblearn.over_sampling import SMOTE\n",
    "\n",
    "oversampling_tf = SMOTE(sampling_strategy='all')\n",
    "\n",
    "oversampling_tf.fit(tfidf,df['label'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'SMOTE' object has no attribute 'fit_sample'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-12-e99a13a15bde>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mfitur_tf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabel_tf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0moversampling_tf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfit_sample\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtfidf\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'label'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfitur_tf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlabel_tf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'SMOTE' object has no attribute 'fit_sample'"
     ]
    }
   ],
   "source": [
    "fitur_tf, label_tf = oversampling_tf.fit_sample(tfidf,df['label'])\n",
    "print(fitur_tf.shape)\n",
    "print(label_tf.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2904, 1807) (1245, 1807)\n"
     ]
    }
   ],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "seed = 1\n",
    "X = df['data_cleaned'].values.astype('U')\n",
    "y = df['label']\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=seed)\n",
    "X_train = vectorizer.fit_transform(X_train) # \"Fit_Transform\"\n",
    "X_test = vectorizer.transform(X_test) # Perhatikan disini hanya \"Transform\"\n",
    "\n",
    "print(X_train.shape, X_test.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2871, 1807) (1232, 1807)\n"
     ]
    }
   ],
   "source": [
    "def hapusKosong(X,y):\n",
    "    y = y[X.getnnz(1)>0] # delete label dokumen yang memiliki row =0 di tfidf-nya\n",
    "    X = X[X.getnnz(1)>0] # Remove Zero Rows\n",
    "    return X, y\n",
    "\n",
    "X_train, y_train = hapusKosong(X_train, y_train)\n",
    "X_test, y_test = hapusKosong(X_test, y_test)\n",
    "print(X_train.shape, X_test.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Akurasi =  0.8222402597402597\n",
      "[[ 34 201]\n",
      " [ 18 979]]\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.65      0.14      0.24       235\n",
      "           1       0.83      0.98      0.90       997\n",
      "\n",
      "    accuracy                           0.82      1232\n",
      "   macro avg       0.74      0.56      0.57      1232\n",
      "weighted avg       0.80      0.82      0.77      1232\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn import svm\n",
    "from sklearn.metrics import classification_report, confusion_matrix, accuracy_score\n",
    "\n",
    "dSVM = svm.SVC( kernel = 'linear')\n",
    "dSVM.fit(X_train, y_train)\n",
    "y_SVM = dSVM.predict(X_test)\n",
    "print('Akurasi = ', accuracy_score(y_test, y_SVM))\n",
    "print(confusion_matrix(y_test, y_SVM))\n",
    "print(classification_report(y_test, y_SVM))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Akurasi =  0.7962662337662337\n",
      "[[ 38 197]\n",
      " [ 54 943]]\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.41      0.16      0.23       235\n",
      "           1       0.83      0.95      0.88       997\n",
      "\n",
      "    accuracy                           0.80      1232\n",
      "   macro avg       0.62      0.55      0.56      1232\n",
      "weighted avg       0.75      0.80      0.76      1232\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn import neighbors\n",
    "from sklearn.metrics import accuracy_score\n",
    "from sklearn.metrics import confusion_matrix, classification_report\n",
    "\n",
    "\n",
    "model = neighbors.KNeighborsClassifier()\n",
    "kNN = model.fit(X_train, y_train)\n",
    "y_kNN = kNN.predict(X_test)\n",
    "\n",
    "print('Akurasi = ', accuracy_score(y_test, y_kNN))\n",
    "print(confusion_matrix(y_test, y_kNN))\n",
    "print(classification_report(y_test, y_kNN))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[137  98]\n",
      " [453 544]]\n",
      "0.5527597402597403\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.23      0.58      0.33       235\n",
      "           1       0.85      0.55      0.66       997\n",
      "\n",
      "    accuracy                           0.55      1232\n",
      "   macro avg       0.54      0.56      0.50      1232\n",
      "weighted avg       0.73      0.55      0.60      1232\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.naive_bayes import GaussianNB\n",
    "naivbayes = GaussianNB()\n",
    "naivbayes.fit(X_train.toarray(), y_train)\n",
    "y_NB = naivbayes.predict(X_test.toarray())\n",
    "print(confusion_matrix(y_test, y_NB))\n",
    "print(accuracy_score(y_test, y_NB))\n",
    "print(classification_report(y_test, y_NB))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 0.8173701298701299\n",
      "[[ 19 216]\n",
      " [  9 988]]\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.68      0.08      0.14       235\n",
      "           1       0.82      0.99      0.90       997\n",
      "\n",
      "    accuracy                           0.82      1232\n",
      "   macro avg       0.75      0.54      0.52      1232\n",
      "weighted avg       0.79      0.82      0.75      1232\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.linear_model import LogisticRegression\n",
    "model_logreg = LogisticRegression(solver=\"lbfgs\", multi_class=\"auto\")\n",
    "model_logreg.fit(X_train, y_train)\n",
    "Y_pred = model_logreg.predict(X_test)\n",
    "print(\"Accuracy:\", accuracy_score(Y_pred, y_test))\n",
    "print(confusion_matrix(y_test, Y_pred))\n",
    "print(classification_report(y_test, Y_pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 0.8198051948051948\n",
      "[[ 41 194]\n",
      " [ 28 969]]\n",
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.59      0.17      0.27       235\n",
      "           1       0.83      0.97      0.90       997\n",
      "\n",
      "    accuracy                           0.82      1232\n",
      "   macro avg       0.71      0.57      0.58      1232\n",
      "weighted avg       0.79      0.82      0.78      1232\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "rf = RandomForestClassifier()\n",
    "rf.fit(X_train, y_train)\n",
    "y_rf = rf.predict(X_test)\n",
    "\n",
    "print(\"Accuracy:\", accuracy_score(y_test, y_rf))\n",
    "print(confusion_matrix(y_test, y_rf))\n",
    "print(classification_report(y_test, y_rf))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: graphviz in c:\\users\\tamu\\anaconda3\\lib\\site-packages (0.14)\n",
      "Requirement already satisfied: dtreeviz in c:\\users\\tamu\\anaconda3\\lib\\site-packages (0.8.2)\n",
      "Requirement already satisfied: numpy in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from dtreeviz) (1.16.5)\n",
      "Requirement already satisfied: colour in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from dtreeviz) (0.1.5)\n",
      "Requirement already satisfied: pandas in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from dtreeviz) (0.25.1)\n",
      "Requirement already satisfied: scikit-learn in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from dtreeviz) (0.24.2)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from dtreeviz) (3.1.1)\n",
      "Requirement already satisfied: pytz>=2017.2 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from pandas->dtreeviz) (2019.3)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from pandas->dtreeviz) (2.8.0)\n",
      "Requirement already satisfied: joblib>=0.11 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from scikit-learn->dtreeviz) (0.13.2)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from scikit-learn->dtreeviz) (2.2.0)\n",
      "Requirement already satisfied: scipy>=0.19.1 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from scikit-learn->dtreeviz) (1.3.1)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from matplotlib->dtreeviz) (0.10.0)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from matplotlib->dtreeviz) (1.1.0)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from matplotlib->dtreeviz) (2.4.2)\n",
      "Requirement already satisfied: six>=1.5 in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from python-dateutil>=2.6.1->pandas->dtreeviz) (1.12.0)\n",
      "Requirement already satisfied: setuptools in c:\\users\\tamu\\anaconda3\\lib\\site-packages (from kiwisolver>=1.0.1->matplotlib->dtreeviz) (41.4.0)\n"
     ]
    }
   ],
   "source": [
    "!pip install graphviz dtreeviz\n",
    "from sklearn.model_selection import cross_val_score\n",
    "import graphviz, pandas as pd, matplotlib.pyplot as plt, numpy as np, seaborn as sns\n",
    "from pandas.plotting import scatter_matrix \n",
    "from sklearn import model_selection, tree\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn import tree\n",
    "from dtreeviz.trees import *\n",
    "from IPython.core.display import display, HTML\n",
    "import numpy as np, matplotlib.pyplot as plt, pandas as pd, seaborn as sns\n",
    "from matplotlib.colors import ListedColormap\n",
    "from sklearn import svm, preprocessing\n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.naive_bayes import GaussianNB\n",
    "from sklearn import neighbors\n",
    "from sklearn.gaussian_process.kernels import RBF\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier\n",
    "from sklearn.datasets import make_blobs, make_moons, make_circles, make_classification\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import classification_report, confusion_matrix, accuracy_score\n",
    "from sklearn.ensemble import VotingClassifier\n",
    "from collections import Counter\n",
    "sns.set(style=\"ticks\", color_codes=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\naive_bayes.py\", line 207, in fit\n",
      "    X, y = self._validate_data(X, y)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\base.py\", line 433, in _validate_data\n",
      "    X, y = check_X_y(X, y, **check_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 878, in check_X_y\n",
      "    estimator=estimator)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 63, in inner_f\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 653, in check_array\n",
      "    accept_large_sparse=accept_large_sparse)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\utils\\validation.py\", line 417, in _ensure_sparse_format\n",
      "    raise TypeError('A sparse matrix was passed, but dense '\n",
      "TypeError: A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array.\n",
      "\n",
      "  FitFailedWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Regresi Logistik [0.8194444444444444, 0.8222996515679443, 0.818815331010453, 0.8292682926829268, 0.8153310104529616, 0.8257839721254355, 0.818815331010453, 0.8222996515679443, 0.818815331010453, 0.8257839721254355]\n",
      "k-NN [0.8159722222222222, 0.8048780487804879, 0.8083623693379791, 0.8118466898954704, 0.8013937282229965, 0.8048780487804879, 0.818815331010453, 0.794425087108014, 0.8013937282229965, 0.8153310104529616]\n",
      "Naive Bayes [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan]\n",
      "Decision Tree [0.7256944444444444, 0.7142857142857143, 0.7177700348432056, 0.710801393728223, 0.7491289198606271, 0.7282229965156795, 0.7177700348432056, 0.6968641114982579, 0.7979094076655052, 0.7073170731707317]\n",
      "Random Forest [0.8159722222222222, 0.794425087108014, 0.8118466898954704, 0.8222996515679443, 0.8362369337979094, 0.8013937282229965, 0.818815331010453, 0.8222996515679443, 0.8118466898954704, 0.8048780487804879]\n",
      "SVM [0.8194444444444444, 0.818815331010453, 0.8118466898954704, 0.8257839721254355, 0.818815331010453, 0.8222996515679443, 0.8153310104529616, 0.8222996515679443, 0.8153310104529616, 0.8222996515679443]\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAssAAAHUCAYAAADMYGUIAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAgAElEQVR4nO3dfVyUdb7/8TcKgjchSNyUbh7d9piltHYwiVJTCFYCUdGySKn2kKcy004KHau1G+/KzUxOZuamdTAlJW6sDCXTLbXWfdSSN2Uu1YmzcaNDByVIYOb3hz/mNMHXAZ1xBF7Px2MeD6+77/W5Zpxr3nznO9flZbPZbAIAAADQTBdPFwAAAABcqAjLAAAAgAFhGQAAADAgLAMAAAAG3p4uoCV1dXU6cOCAgoOD1bVrV0+XAwAAgA6qsbFRlZWVGjJkiPz8/JotvyDD8oEDB5SSkuLpMgAAANBJZGVlKSIiotn8CzIsBwcHSzpddFhYmIerAQAAQEdVVlamlJQUe/78pQsyLDcNvQgLC1O/fv08XA0AAAA6OtPQX37gBwAAABgQlgEAAAADwjIAAABg0KqwXFBQoPj4eMXGxiorK6vZ8oMHDyo5OVnjx4/XjBkzVF1d7bC8rKxM1157rUpLS11TNQAAAHAeOA3L5eXlWr58uTZs2KDc3Fxt2rRJR48edVhn4cKFmjVrlvLz8zVgwACtXbvWvsxqtWr+/Pmqr693ffUAAACAGzkNy3v27FFkZKQCAgLUo0cPxcXFadu2bQ7rWK1W1dTUSJJqa2sdLuj8yiuvKCoqSoGBgS22X11drdLSUodHWVnZuRwTAAAA4BJOLx1XUVHhcN25kJAQFRcXO6yTkZGhu+++W4sWLVL37t2VnZ0t6fTNRfbt26dXXnmlxeEbkrR+/XplZmaeyzEAAAAAbuE0LFutVnl5edmnbTabw3RdXZ3mz5+vdevWKTw8XK+++qrS09O1YsUKPfHEE1qxYoW6dDF3YKempmrixIkO85ouDg0AAAB4ktOwHBYWpv3799unKysrFRISYp8+cuSIfH19FR4eLkm69dZbtWLFCu3fv1/Hjx/XvffeK+l0D/U999yjzMxMDRw40L69v7+//P39XXZAAAAAgKs4HbMcFRWlvXv3ymKxqLa2VoWFhRo1apR9ef/+/VVWVqaSkhJJUlFRkYYOHaqRI0fq/fffV15envLy8hQSEqKXX37ZISgDAAAAFzKnPcuhoaGaM2eOpk+frvr6ek2ePFnh4eFKS0vTrFmzNHToUC1evFizZ8+WzWZTUFCQFi1adD5qBwAAANzKy2az2TxdxC+VlpYqOjpaRUVF6tevn6fLAQAAQAflLHdyBz8AAADAgLAMAAAAGBCWAQAAAAPCMgAAAGDg9GoYAACgcygqKlJhYaFb2q6qqpIkBQYGurzt2NhYRUdHu7xdQKJnGQAAnAdVVVX2wAy0J/QsAwAASVJ0dLTbemjT09MlSUuXLnVL+4C70LMMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABh4e7qAC1lRUZEKCwtd3m5VVZUkKTAw0OVtS1JsbKyio6Pd0jYAAEBnQs+yB1RVVdkDMwAAAC5c9CyfQXR0tFt6aNPT0yVJS5cudXnbAAAAcB16lgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgOssAwAAdADcedg96FkGAACAUWe/83CH6FlevXq1SkpKPF1GqzXV2nQnv/Zi4MCBmjFjhqfLAAAALeDOw+7RIcJySUmJPj/4hbr6uefrAVezNpzu0D/093IPV9J6jXWd9y9KAADQeXWIsCxJXf0CddHAmzxdRod1omS7p0sAAAA47xizDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgEGHuRoGAABAe8D9Ic4PV90fgrAMAABwHpWUlOjwwQPq1b2bp0tpFVt9oyTpu5IjHq6k9U7WnnJZW4RlAADamfbWMym1z95Jd965tlf3boq4PMQtbUPaf7TCZW0RlgEAaGdKSkp06NCX6tm9j6dLabXGhtOR49uvKz1cSevU1Fo8XQIuEK0KywUFBVq1apUaGhqUmpqqlJQUh+UHDx7U448/rvr6el1yySV69tln5e/vr7///e96/PHHdfLkSfn5+WnBggUaPHiwyw+iqqpKjXVV3GXOjRrrqlRV1T6+LgKAzqBn9z4K/028p8vosIq/esfTJeAC4fRqGOXl5Vq+fLk2bNig3Nxcbdq0SUePHnVYZ+HChZo1a5by8/M1YMAArV27VpL06KOPKi0tTXl5eZo9e3a7+uoFAAAAcNqzvGfPHkVGRiogIECSFBcXp23btmnmzJn2daxWq2pqaiRJtbW16t27tyRpypQpGjlypCRp0KBB+v7775u1X11drerqaod5ZWVlbTqIwMBAfW85pYsG3tSm7dB6J0q2KzAw0NNlAAAAnFdOw3JFRYWCg4Pt0yEhISouLnZYJyMjQ3fffbcWLVqk7t27Kzs7W5I0adIk+zovvPCCYmJimrW/fv16ZWZmnvUBAAAAAO7iNCxbrVZ5eXnZp202m8N0XV2d5s+fr3Xr1ik8PFyvvvqq0tPT9fLLL9vXf+aZZ/S3v/1Nr732WrP2U1NTNXHiRId5ZWVlzcZFAwAAAOeb07AcFham/fv326crKysVEvJ/lzo5cuSIfH19FR4eLkm69dZbtWLFCklSQ0OD0tPTVV5ertdee00XXXRRs/b9/f3l7+9/zgcCAADQHlRVVelE7SmXXt4Mjk7UnlJVVZVL2nIalqOiorRy5UpZLBZ1795dhYWFeuqpp+zL+/fvr7KyMpWUlGjgwIEqKirS0KFDJUlLly7VyZMn9ac//Undurn3Sgrt6WoY1oZaSVIX7+4erqT1GuuqJIV6uow2KSoqUmFhocvbbXrzuWMMd2xsrKKjo13eLgAAODtOw3JoaKjmzJmj6dOnq76+XpMnT1Z4eLjS0tI0a9YsDR06VIsXL9bs2bNls9kUFBSkRYsWyWKxKCsrS/369dOUKVPs7eXl5bn8IAYOHOjyNt2p6cLsAwe2p/AZ2u6eZ3dxZ1gGAHR8gYGBOllVyU1J3Gj/0QqXfU636jrLiYmJSkxMdJi3Zs0a+79Hjx6t0aNHN9vu0KFD51he67jr7jru0nQJvaVLl3q4ko4tOjraLb20vH4AAHQeTq+zDAAAAHRWhGUAAADAgLAMAAAAGBCWAQAAAINW/cAPAAAArnOyHV1n+VR9oySpm09XD1fSeidrT7msLcIyAADAedTeLsXadMnbX7Wzul31PBOWz8BdN7Vo+k/XdAkyV+PGFgAAXLi45G37Qlj2AG5mAQAA0D4Qls/AXTe1AAAAQPtAWIZHrV692j4spb1w9zAadxg4cGC7+9oPAIALAWEZHlVSUqKvDh1QWC8fT5fSat2tp38VfOK/v/RwJa1TdrLe0yUAcLGqqirV/HhcxV+94+lSOqyaH4+rqoqYBMIyLgBhvXz0+2EXe7qMDmvtp8c8XQIAAO0WYRkAgHYmMDBQ1T80KPw38Z4upcMq/uodfpAPSdzBDwAAADAiLAMAAAAGhGUAAADAgDHLAAAAHQB3HnYPwjIAAACMOvsPHQnLAAAAHQB3HnYPxiwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAANvTxeAzq2qqkqVJ+u19tNjni6lw/r+ZL0aqqo8XQYAF6uptaj4q3c8XUarnaqvlSR18+nu4Upap6bWIinY02XgAkBYBgCgnRk4cKCnS2izkpISSVL/Ae0lgAa3y+cZrkdYhkcFBgbK+0SFfj/sYk+X0mGt/fSYLgoM9HQZAFxoxowZni6hzdLT0yVJS5cu9XAlQNswZhkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAG3O4aHld2sl5rPz3m6TJa7eSpRklSr25dPVxJ65SdrNdFni4CAIB2irAMjxo4cKCnS2izypISSdIll7WP2i9S+3yeAQC4EBCW4VEzZszwdAltlp6eLklaunSphysBAADuxphlAAAAwKBVYbmgoEDx8fGKjY1VVlZWs+UHDx5UcnKyxo8frxkzZqi6ulqSVF1drXvuuUfjxo1TSkqKKisrXVs9AAAA4EZOw3J5ebmWL1+uDRs2KDc3V5s2bdLRo0cd1lm4cKFmzZql/Px8DRgwQGvXrpUkPf/884qIiNC7776rKVOmaOHChe45CgAAAMANnIblPXv2KDIyUgEBAerRo4fi4uK0bds2h3WsVqtqamokSbW1tfLz85MkffDBB0pMTJQkJSQkaPfu3aqvr3fYtrq6WqWlpQ6PsrIylxwcAAAAcC6c/sCvoqJCwcHB9umQkBAVFxc7rJORkaG7775bixYtUvfu3ZWdnd1sW29vb/Xq1UsWi0WhoaH2bdevX6/MzEyXHAwAAADgSk7DstVqlZeXl33aZrM5TNfV1Wn+/Plat26dwsPD9eqrryo9PV0vv/xys7ZsNpu6dHHszE5NTdXEiRMd5pWVlSklJaXNBwMAAAC4ktNhGGFhYQ4/zKusrFRISIh9+siRI/L19VV4eLgk6dZbb9Unn3wi6XQv9LFjp2820dDQoJqaGgUEBDi07+/vr379+jk8wsLCzv3IAAAdisVi0bx582SxWDxdCoBOxGlYjoqK0t69e2WxWFRbW6vCwkKNGjXKvrx///4qKytTyf+/UUNRUZGGDh0qSRo9erRyc3MlSe+8844iIiLk4+PjjuMAAHRwGzZs0MGDB/XGG294uhQAnYjTYRihoaGaM2eOpk+frvr6ek2ePFnh4eFKS0vTrFmzNHToUC1evFizZ8+WzWZTUFCQFi1aJEl68MEHlZGRoZtvvlkXXXSRli1b5vYDAgB0PBaLRTt27JDNZtP27dt12223qU+fPp4uC0An0Ko7+CUmJtqvatFkzZo19n+PHj1ao0ePbrZdQECAXnrppXMsEQDQ2W3YsEFWq1XS6d/SvPHGG7r//vs9XBWAzoA7+AEALngffPCBGhoaJJ3+DczOnTs9XBGAzoKwDAC44N14443y9j79Zai3t7fGjBnj4YoAdBaEZQDABe/222+3X3q0S5cuuu222zxcEYDOgrAMALjg9enTRzExMfLy8tJNN93Ej/sAnDet+oEfAACedvvtt+u///u/6VUGcF4RlgEA7UKfPn30zDPPeLoMAJ0MwzAAAAAAA8IyAAAAYEBYBgAAAAwYs4wOq6ioSIWFhS5vt6SkRJKUnp7u8rZjY2MVHR3t8nYBoDXcdd6UOHei/SIsA20UGBjo6RIAoN3h3In2irCMDis6OpqeBgBoA86bQHOMWQYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAw8G7NSgUFBVq1apUaGhqUmpqqlJQU+7LDhw8rIyPDPm2xWNS7d29t3bpVpaWlSk9P18mTJ+Xv768lS5aob9++rj8KAAAAwA2c9iyXl5dr+fLl2rBhg3Jzc7Vp0yYdPXrUvnzw4MHKy8tTXl6eNm7cqN69e2vBggWSpBUrVujmm29WXl6eYmNjtXz5crcdCAAAAOBqTsPynj17FBkZqYCAAPXo0UNxcXHatm1bi+uuXr1aw4cPV0REhCTJarXq5MmTkqTa2lr5+fm5sHQAAADAvZwOw6ioqFBwcLB9OiQkRMXFxc3WO3HihLKzs1VQUGCf9+CDD2rq1Kl6/fXXVV9fr02bNjXbrrq6WtXV1Q7zysrK2nQQAAAAgDs4DctWq1VeXl72aZvN5jDdJD8/XzExMQoKCrLPS09P15NPPqmYmBi99957mjlzpvLz8x22X79+vTIzM8/1OAAAAACXczoMIywsTJWVlfbpyspKhYSENFtvx44dio+Pt09bLBaVlJQoJiZGkhQXF6fKykpVVVU5bJeamqqioiKHR1ZW1lkfEAAAAOAqTsNyVFSU9u7dK4vFotraWhUWFmrUqFEO69hsNh08eFDDhg2zzwsMDJSvr6/2798vSfrrX/+qnj17qk+fPg7b+vv7q1+/fg6PsLAwVxwbAAAAcE6cDsMIDQ3VnDlzNH36dNXX12vy5MkKDw9XWlqaZs2apaFDh8piscjHx0e+vr727by8vJSZmamnnnpKdXV16tmzp1auXOnWgwEAAABcyctms9k8XcQvlZaWKjo6WkVFRerXr5+nywEAAEAH5Sx3cgc/AAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABt6tWamgoECrVq1SQ0ODUlNTlZKSYl92+PBhZWRk2KctFot69+6trVu3qqKiQo8++qgqKirk5+enZcuWqV+/fq4/CgAAAMANnPYsl5eXa/ny5dqwYYNyc3O1adMmHT161L588ODBysvLU15enjZu3KjevXtrwYIFkqR58+ZpzJgxys3NVVJSkpYtW+a2AwEAAABczWnP8p49exQZGamAgABJUlxcnLZt26aZM2c2W3f16tUaPny4IiIiZLFY9MUXX+jVV1+VJCUnJ+u6665rtk11dbWqq6sd5pWVlZ3VwQAAAACu5DQsV1RUKDg42D4dEhKi4uLiZuudOHFC2dnZKigokCR99913uvTSS7VkyRLt379fwcHBeuyxx5ptt379emVmZp7LMQAAAABu4XQYhtVqlZeXl33aZrM5TDfJz89XTEyMgoKCJEkNDQ06dOiQIiMjtWXLFkVHRzuMbW6SmpqqoqIih0dWVta5HBMAAADgEk7DclhYmCorK+3TlZWVCgkJabbejh07FB8fb58ODg5Wz549NWbMGElSQkJCiz3S/v7+6tevn8MjLCzsrA4GAAAAcCWnYTkqKkp79+6VxWJRbW2tCgsLNWrUKId1bDabDh48qGHDhtnnXXbZZQoLC9OuXbskSTt37tRVV13l4vIBAAAA93EalkNDQzVnzhxNnz5dEyZMUEJCgsLDw5WWlqbPP/9c0unLxfn4+MjX19dh25UrV+qVV15RQkKCXnvtNS1atMg9RwEAAAC4gZfNZrN5uohfKi0tVXR0tIqKirguMwAAANzGWe7kDn4AAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMvD1dAACgYykqKlJhYaHL262qqpIkBQYGurzt2NhYRUdHu7xdAO0fPcsAgHahqqrKHpgB4HyhZxkA4FLR0dFu6aVNT0+XJC1dutTlbQOACT3LAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABi0KiwXFBQoPj5esbGxysrKclh2+PBhJSUl2R8jR45UQkKCwzqHDh3SkCFDXFc1AAAAcB54O1uhvLxcy5cvV05Ojrp166apU6dqxIgRuvzyyyVJgwcPVl5eniSptrZWU6ZM0YIFC+zb19bW6qmnnlJ9fb17jgAAAABwE6c9y3v27FFkZKQCAgLUo0cPxcXFadu2bS2uu3r1ag0fPlwRERH2eUuWLFFqaqqx/erqapWWljo8ysrKzuJQAAAAANdy2rNcUVGh4OBg+3RISIiKi4ubrXfixAllZ2eroKDAPq+oqEh1dXX63e9+Z2x//fr1yszMbGvdAAAAgNs5DctWq1VeXl72aZvN5jDdJD8/XzExMQoKCpIkVVZWatWqVVq3bt0Z209NTdXEiRMd5pWVlSklJaU19QMAAABu4zQsh4WFaf/+/fbpyspKhYSENFtvx44dmjFjhn36gw8+0A8//OAQepOSkpSVlaVevXrZ5/n7+8vf3/+sDwAAAABwF6djlqOiorR3715ZLBbV1taqsLBQo0aNcljHZrPp4MGDGjZsmH3elClTtGPHDuXl5dl/AJiXl+cQlAEAAIALmdOwHBoaqjlz5mj69OmaMGGCEhISFB4errS0NH3++eeSJIvFIh8fH/n6+rq9YAAAAOB8cToMQ5ISExOVmJjoMG/NmjX2fwcFBemjjz46YxtffvnlWZQHAAAAeA538AMAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAbeni4AAHD+rV69WiUlJZ4uo02a6k1PT/dwJW0zcOBAzZgxw9NlADhLhGUA6IRKSkr05YEDurhr+/kY8LFaJUnHD3/h4Upa71hjg6dLAHCO2s9ZEgDgUhd39dakgD6eLqNDy/nB4ukSAJwjxiwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAberVmpoKBAq1atUkNDg1JTU5WSkmJfdvjwYWVkZNinLRaLevfura1bt+qvf/2rFi9erPr6egUEBGjRokXq27ev648CAAAAcAOnYbm8vFzLly9XTk6OunXrpqlTp2rEiBG6/PLLJUmDBw9WXl6eJKm2tlZTpkzRggULJElz587Viy++qCuuuEKbN2/W008/rVWrVrnvaAAAAAAXcjoMY8+ePYqMjFRAQIB69OihuLg4bdu2rcV1V69ereHDhysiIkKnTp3Sgw8+qCuuuEKSNGjQIH3//ffNtqmurlZpaanDo6ys7BwPCwAAADh3TnuWKyoqFBwcbJ8OCQlRcXFxs/VOnDih7OxsFRQUSJK6deumpKQkSZLValVmZqZiYmKabbd+/XplZmae9QEAAAAA7uI0LFutVnl5edmnbTabw3ST/Px8xcTEKCgoyGH+qVOnlJGRoYaGBs2YMaPZdqmpqZo4caLDvLKyModx0QAAAIAnOA3LYWFh2r9/v326srJSISEhzdbbsWNHszBcU1Oje++9VwEBAVq1apV8fHyabefv7y9/f/+zqR0AAABwK6djlqOiorR3715ZLBbV1taqsLBQo0aNcljHZrPp4MGDGjZsmMP8uXPnqn///nr++efVrVs311YOAAAAuJnTnuXQ0FDNmTNH06dPV319vSZPnqzw8HClpaVp1qxZGjp0qCwWi3x8fOTr62vf7tChQyoqKtLll19uH2YREhKiNWvWuO9oAAAAABdq1XWWExMTlZiY6DDv56E3KChIH330kcPyK6+8Ul9++aULSgQAuFpVVZWONTQo5weLp0vp0I41NKhLVZWnywBwDriDHwAAAGDQqp5lAEDHEhgYKGtZuSYF9PF0KR1azg8WBQYGeroMAOeAnmUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAICBt6cLAAB4xrHGBuX8YPF0Ga32o9UqSerRpf308xxrbFCQp4sAcE4IywDQCQ0cONDTJbTZ/5aUSJKC2lHtQWqfzzWA/0NYBoBOaMaMGZ4uoc3S09MlSUuXLvVwJQA6k/bzXRYAAABwnhGWAQAAAAPCMgAAAGBAWAYAAAAMCMsAAACAAWEZAAAAMCAsAwAAAAaEZQAAAMCAsAwAAAAYEJYBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAAPCMgAAAGBAWAYAAAAMvFuzUkFBgVatWqWGhrJmFpgAABU5SURBVAalpqYqJSXFvuzw4cPKyMiwT1ssFvXu3Vtbt27VP/7xD82dO1fHjx/XgAEDtGzZMvXs2dP1RwEAAAC4gdOe5fLyci1fvlwbNmxQbm6uNm3apKNHj9qXDx48WHl5ecrLy9PGjRvVu3dvLViwQJL0xBNP6Pbbb9e2bds0ZMgQvfjii247EAAAAMDVnIblPXv2KDIyUgEBAerRo4fi4uK0bdu2FtddvXq1hg8froiICNXX1+svf/mL4uLiJEmTJk0ybgcAAABciJwOw6ioqFBwcLB9OiQkRMXFxc3WO3HihLKzs1VQUCBJqqqqUq9eveTtfXoXwcHBKi8vb7ZddXW1qqurHeaVlZW17SgAAAAAN3Aalq1Wq7y8vOzTNpvNYbpJfn6+YmJiFBQUZFyvpe3Wr1+vzMzMNhcOAAAAuJvTsBwWFqb9+/fbpysrKxUSEtJsvR07dmjGjBn26T59+ujEiRNqbGxU165djdulpqZq4sSJDvPKysocfkQIAAAAeILTMctRUVHau3evLBaLamtrVVhYqFGjRjmsY7PZdPDgQQ0bNsw+z8fHRxEREXrnnXckSbm5uc22kyR/f3/169fP4REWFnauxwUAAACcM6dhOTQ0VHPmzNH06dM1YcIEJSQkKDw8XGlpafr8888lnb5cnI+Pj3x9fR22/cMf/qDs7GzFx8dr//79mj17tnuOAgAAAHCDVl1nOTExUYmJiQ7z1qxZY/93UFCQPvroo2bb9e3bV6+//vo5lggAAAB4BnfwAwAAAAwIywAAAIBBq4ZhAADQWkVFRSosLHR5uyUlJZKk9PR0l7cdGxur6Ohol7cLoP0jLAMA2oXAwEBPlwCgEyIsAwBcKjo6ml5aAB0GY5YBAAAAA8IyAAAAYEBYBgAAAAwIywAAAIABYRkAAAAwICwDAAAABoRlAAAAwICwDAAAABgQlgEAAAADwjIAAABgQFgGAAAADAjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADAgLAMAAAAGBCWAQAAAANvTxfQksbGRklSWVmZhysBAABAR9aUN5vy5y9dkGG5srJSkpSSkuLhSgAAANAZVFZWqn///s3me9lsNpsH6jmjuro6HThwQMHBweratauny3G5srIypaSkKCsrS2FhYZ4uB23E69d+8dq1b7x+7RuvX/vV0V+7xsZGVVZWasiQIfLz82u2/ILsWfbz81NERISny3C7sLAw9evXz9Nl4Czx+rVfvHbtG69f+8br13515NeupR7lJvzADwAAADAgLAMAAAAGhGUAAADAoOuCBQsWeLqIzsjX11cjRoyQr6+vp0vBWeD1a7947do3Xr/2jdev/erMr90FeTUMAAAA4ELAMAwAAADAgLAMAAAAGBCWAQAAAIMOHZZLS0s1ZMgQJSUlKSkpSYmJiRo7dqxeeOEFj9Qzf/58ff755w7zPv74Y02bNs0l7SclJZ1x+c/307TuypUrtXLlSpfsvz1rzeswduxYLV++3GFeRkaGcnJyWrUcjkpLSzVo0CB99NFHDvPHjh2r0tJS43bl5eVKS0s75/0PGjTIfm6Ij4/X/Pnz9dNPP51zux3VL8+ncXFxeuSRR3Ts2LGzam/FihUqKioyLm/pfNlWW7Zssdc7ZMgQxcfHKykpSU888cQ5tdseuPvzLycnRxkZGS5p6+dtXnvttfaak5KS9Pvf/96l+/i54uJiPfvss25rv73Ytm2bJk2apPHjxysxMVGvvPKK3nzzzRaf+0ceeUSvvfaacnJyNGjQIG3dutVh+bp16zRo0KAznsPbowvyDn6uFBISory8PPt0eXm54uLidPPNN+vXv/71ea1l4cKFbm3/58fZkk8++aTV66Jl69ev10033aQhQ4ac1XI48vHx0WOPPab8/Hz16tWrVduEhoZqzZo1Ltl/0/vAZrPpgQce0ObNm5WSkuKStjuin59PbTabnnvuOc2aNUsbNmxoc1sPPvjgGZe74nyZnJys5ORkSaf/CHv55Zc77N3HWnIhff611tixY7VkyZLzsq+jR4/q+PHj52VfF6ry8nItXbpUOTk5CgwMVE1NjaZNm6b77rtPn332mY4fP66goCBJUm1trXbu3Kl58+Zp586dCgsL03vvvaeEhAR7e9u3b5e/v7+nDsdtOnTPcksqKytls9nUs2dPSdLLL7+siRMnavz48XrmmWfUdHGQ1157TbGxsUpOTtbcuXPtva+RkZH613/9VyUlJam+vr7F7U+ePKl77rlHkyZN0qRJk+y9J9OmTdPHH3/c6lpfeuklxcfHKzExUUuWLFFjY+MZaxs0aJAkae/evfZ933XXXbJYLHr66aclSVOmTHFYt0ljY6NmzZqlZ5555qye145k/fr1mjZtmmpra5stmzFjhh555BGdOnWqxW2dLYejkJAQRUVFaenSpc2WNTQ06NFHH9Wtt96q6Oho3Xfffaqrq1NpaanGjh2rqqoqXX/99aqvr5ckHTlyROPHj5ck5ebmauLEiUpKStJ//Md/OO0xrq+vV21trS6++GJ7W9OmTVNycrLGjBmjN954Q1arVWPHjtXXX38tSfrxxx81evRo/fTTT9q9e7cmT56sCRMmaObMmaqqqpIkLV26VOPHj9eECROUmZnpsuftQuDl5aUHHnhAX331lb744gtJ5vPpunXrFBcXp/j4eHtPXtO3Lq05X7Z0LiwtLdWECRM0d+5cJSQkKDU1VT/88EOr6//44481efJkTZo0Senp6aqpqVF6eromTZqkpKQke49ZY2OjFi9ebD+udevWueopPO9+/vl3pveX6XnNzc1VXFyckpOT9cEHH9jb/eyzzzRlyhSNHz9eqamp+vbbbyWdfg0XL16syZMnKzExUbt27dLvf/973XjjjW1+Hs+0j5kzZyouLk6HDx9u9XuxurpaL7zwgt5//32tWrXq3J/cdqqqqkr19fWqq6uTJPXs2VNLlizRb37zG8XExOidd96xr7tjxw5FRkYqMDBQkjR8+HAdOHBAP/74oyTpH//4h3r27KmLLrro/B+Im3X4sFxRUaGkpCT97ne/04gRI/T8888rMzNTYWFh2r17tw4cOKDNmzcrNzdX5eXlys/P1xdffKGsrCzl5ORow4YN9jeldPo/VlpamvLy8rR3794Wt9++fbv69u2rnJwcLVy4UPv3729z3bt27dL777+vLVu26K233tK3336rjRs3nrG2Ji+++KIWLFignJwcRUVF6dChQ3r00UclSW+++Waz9W02mx599FGFhYVp3rx5ba61I8nJyVFhYaFeeuklde/evdnyxMRE/epXv9J//ud/tri9s+VoLiMjQx9++GGz4RiffvqpfHx8tGnTJm3fvl0nTpzQrl277MsDAwMVHh6uDz/8UJL09ttva/z48frqq6+UnZ2tjRs3Ki8vT0FBQVq7dm2L+276qnfkyJGqrKzUddddJ+n0++S+++7Tli1b9Nprr+mZZ55Rly5dNGHCBOXn50uSCgsLdeONN6qmpkZ//OMftXbtWuXm5uqGG27QsmXL9D//8z/avXu38vPz9cYbb+jo0aMdbphHt27d1L9/f5WUlBjPp8XFxdqwYYM2b96s/Px8HTx4UAcOHLC34ex8aToXStIXX3yhu+66S1u3bpW/v78KCgraVP8333yj9evXa+nSpVq1apWuuuoq5eTkKCsrSy+99JK+++47ZWdnS5Leeustbd68WUVFRWd1TveEM33+nen91dLzWl5ermXLlikrK0ubNm1STU2NJOnUqVN66KGH7N8QTZ06VQ899JC9BpvNps2bNysuLk5PP/20MjMzlZWVZTxHvv/++w7DMPbt2+d0H4MGDdJ7772n0NDQVr8XfX19NWvWLI0dO1b33nuvG1+FC9sVV1yh6OhoxcTEaPLkyXr22WdltVrVv39/JScnOwyzyM3N1eTJk+3T3t7euuGGG+z/b9555x2NGzfuvB/D+dBphmFYrVYtWbJEf//733X99ddLOt0DW1xcrEmTJkmS6urqdOmll8pisWjMmDH2r4VvvvlmVVdX29u8+uqrz7h9cnKynnvuOZWXl+vGG2/U/fff3+a69+3bp5tvvtke2JKTk5Wbm6tTp06dsTZJio6O1syZMxUTE6Po6Gj78Zps3LhRJ06cOOP4wc7gyJEjeuyxx/Tcc8/Zv3loyRNPPKGkpCTddNNNZ7Ucjnr16qWnnnrK/kHYZPjw4QoICFBWVpZKSkr0zTff2HswmowfP15vv/22xowZo3fffVevv/66tm/frm+//Va33HKLpNO9xldeeWWL+276itpqtWrRokWaM2eO1q5dq4yMDP35z3/W6tWrdeTIEft+m76tefDBB/XWW2/poYce0t/+9jd9//33mj59ur2t3r17KzQ0VL6+vpo6darGjBmjhx9+uENezN/Ly0t+fn7G8+GxY8c0ZswYe2/TL3sUhw0bdsbzpelcOHr0aAUFBdlf29/85jf63//93zbVPmDAAHtde/bsUV1dnbZs2SLp9DcHX331lfbu3avDhw9r37599vlffvmlIiIi2rQvTzjT59+Z3l8tPa+ffvqphg0bZv/2JTExUfv27dM333wjf39/hYeHS5LGjRunxx9/XCdOnJAkjRo1SpJ06aWX6uqrr1b37t3Vt2/fZp9bTVoahnHkyJEz7qNpfmd/L56tJ554Qvfdd58+/PBDffjhh7rlllu0bNky3XTTTaqqqtJ3330nPz8/ffPNN4qKinLYdty4ccrOzta4ceO0Y8cOrVmzpkP+DqrDh+UmXbp00bx58zRhwgStXbtWaWlpamxsVGpqqu666y5JUnV1tbp27arNmzfLarUa2/Lz85Mk4/Y9e/bUu+++qz//+c/auXOn/vSnPzl8ldEaLe2/oaFBXbp0OWNtknTnnXdqzJgx2rlzp5599lkVFxef8S/nYcOG6corr9TTTz/tsR8/Xgh69uypxYsXa+HChfrtb3+re+65R9LpD5yfj5ENDg5WRkaGHnnkEf3zP/9zs3acLUdzN9xwQ7PhGEVFRXrhhRc0ffp0TZo0SVVVVfrlPZSio6O1ZMkS/eUvf9Ell1yi0NBQNTY2aty4cfZvU2pqauxDmEy6dOmiyZMn67bbbpMkzZ49W/7+/hozZozi4+PtvSv9+vXTpZdeqsLCQh0/flxXX321duzYoWuuuUYvvfSSJOmnn35STU2NvL299eabb+qTTz7R7t27NXXqVL3++usaMGCAy543Tzt16pS+/vprXX755dq3b5/xfOrl5WXfpry83OFbm3/6p3864/nSdC6U5BB4vLy8mv3/cKbpXN60n2effVZXXXWVJOnYsWPq3bu3tmzZorlz5yo2NlaSZLFYzvjH9IWopc+/M72/Wnpef/n8enufjg8tvT42m83+nvPx8Wm2TVs528fPP5Pb8l6E9MEHH+jHH39UfHy8fYx/dna2Nm/erNjYWE2YMEFbt26Vn5+fkpKS1KWL44CEESNG6LHHHtORI0cUGBjYIYdgSJ1gGMbPeXt7a968eXrxxRdVWVmpyMhI5eXlqaamRg0NDbr//vv13nvv6brrrtOuXbt08uRJnTp1SoWFhQ4n+yam7f/rv/5LK1eu1Lhx4/SHP/xBFotFJ0+ebFOtkZGRevvtt1VXV6eGhgZt2bJFkZGRraptypQpqqmp0Z133qk777xThw4dkiR17drV/iHzc1dccYXS0tL01Vdf6f33329TnR1J3759NXbsWF177bV69dVXlZeXp7y8vBZ/TDZ+/Hj96le/0nvvvddiW86Wo7mm4RgVFRWSTn9zM27cOCUnJ8vf318ff/xxs9DbrVs3jRw5UosWLbKPVx4xYoS2b9+u48ePy2azacGCBVq/fr3T/e/du9fem/bRRx9p1qxZiomJ0e7duyXJvu/k5GQ9/fTT9v1dffXV+uyzz+xjmV988UU988wzOnTokO644w4NHz5c6enp+vWvf21fpyOwWq1auXKlrr76al122WXG82FERIR27dpln//v//7vDsMwnJ0vTedCV4uMjNQbb7wh6fTwhfHjx+v7779XZGSksrOzVV9fr5qaGt1+++367LPPXL5/d/vl519r3l8/9y//8i/67LPPVF5eLqvVav+DZuDAgfrhhx9UXFws6fRX8ZdeeqkCAgJcVntr99HW96LpM7Ez8fPz0x//+Ef71StsNpsOHz6swYMHS5ImTpyo7du326+Y8Utdu3bV9ddfr8cff1zx8fHntfbzqdP0LDcZNWqUhg0bphUrVujpp5/WF198oVtuuUWNjY0aOXKkJk6cKC8vL02fPl233nqrevToocDAwBa/shk7dmyL29fU1Oihhx5SYmKiunbtqrlz557x16H79+/XsGHD7NOJiYl68skndfjwYSUnJ6uhoUE33HCD7rjjDnl7ezut7aGHHlJGRoa8vb3Vo0cP+4/7oqOjlZSU1OKlzLp166YFCxYoIyNDI0aMaHc9J640b948JSQkKDEx0d7L1JInnnjC4VfAbV0OR03DMZouVzRlyhQ9/PDDevvtt+Xj46NrrrmmxcsRJSUlKT8/X3FxcZJO//E3c+ZMpaamymq1avDgwfZvCVraVjrde3bRRRfpySeflCQ98MADuv322+Xr66srrrhCffv2VWlpqfr376/Y2Fg99thj9m2Dg4O1aNEizZ49W1arVaGhoXr22WcVGBio3/72t0pISFD37t11zTXX2L+Sbq+axsBKsj+3zz33nCTz+dDLy0t33HGHpk6dKqvVqptuuklRUVH2ITcTJkw44/lyzJgxLZ4Ly8rKXHpsM2fO1IIFC5SQkKDGxkbNnTtXl112maZOnapvv/1WEydOVENDgyZNmqQRI0a4dN/ny88//6ZNm9aq91eTiy++WI8++qjuvPNOde/eXZdffrmk058dy5cv11NPPaXa2lr17t272SU0z1Vr99HW9+J3332nzMxMLVu2TA8//LBLa24vIiMjNXPmTP3bv/2b/cfSI0eOtA+HuuSSSxQYGCir1Wq8ksy4ceOUl5ensWPHnre6zzcvW1u/t+oEvv76a+3atUt33nmnJOnee+/VlClTLoj/CBdybUBHZ7PZtHv3br3xxhv2r3oBAB1bp+tZbo2+ffvq888/V0JCgry8vHTDDTdozJgxni5L0oVdG9DRLVq0SDt37nTZdZ4BABc+epYBAAAAg071Az8AAACgLQjLAAAAgAFhGQAAADAgLAMAAAAGhGUAAADA4P8BhLSngCoT7bUAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 864x576 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "clf = LogisticRegression(solver='liblinear')\n",
    "\n",
    "kNN = neighbors.KNeighborsClassifier()\n",
    "gnb = GaussianNB()\n",
    "dt = tree.DecisionTreeClassifier()\n",
    "rf = RandomForestClassifier()\n",
    "svm_ = svm.SVC()\n",
    "Models = [('Regresi Logistik', clf), ('k-NN',kNN), ('Naive Bayes',gnb), ('Decision Tree', dt), ('Random Forest', rf), ('SVM', svm_)]\n",
    "Scores = {}\n",
    "for model_name, model in Models:\n",
    "    Scores[model_name] = cross_val_score(model, X_train, y_train, cv=10, scoring='accuracy')\n",
    "\n",
    "dt = pd.DataFrame.from_dict(Scores)\n",
    "plt.figure(figsize=(12,8))\n",
    "ax = sns.boxplot(data=dt)\n",
    "for m, s in Scores.items():\n",
    "    print(m, list(s)[:10])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# cross validasi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py:619: FitFailedWarning: Estimator fit failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_validation.py\", line 598, in _fit_and_score\n",
      "    estimator.fit(X_train, y_train, **fit_params)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 1306, in fit\n",
      "    solver = _check_solver(self.solver, self.penalty, self.dual)\n",
      "  File \"C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py\", line 444, in _check_solver\n",
      "    \"got %s penalty.\" % (solver, penalty))\n",
      "ValueError: Solver lbfgs supports only 'l2' or 'none' penalties, got l1 penalty.\n",
      "\n",
      "  FitFailedWarning)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "best parameters {'C': 1.0, 'penalty': 'l2'}\n",
      "accuracy : 0.8220141308555944\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\linear_model\\_logistic.py:765: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)\n",
      "C:\\Users\\tamu\\Anaconda3\\lib\\site-packages\\sklearn\\model_selection\\_search.py:925: UserWarning: One or more of the test scores are non-finite: [       nan 0.81469948        nan 0.81469948        nan 0.81469948\n",
      "        nan 0.82201413        nan 0.79901641        nan 0.75756751\n",
      "        nan 0.74468036]\n",
      "  category=UserWarning\n"
     ]
    }
   ],
   "source": [
    "# Grid search cross validation\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "grid={\"C\":np.logspace(-3,3,7), \"penalty\":[\"l1\",\"l2\"]}# l1 lasso l2 ridge\n",
    "logreg=LogisticRegression()\n",
    "logreg_cv=GridSearchCV(logreg,grid,cv=10)\n",
    "logreg_cv.fit(X_train,y_train)\n",
    "\n",
    "print(\"best parameters\",logreg_cv.best_params_)\n",
    "print(\"accuracy :\",logreg_cv.best_score_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "best parameters  {'criterion': 'entropy', 'max_depth': 100, 'max_features': 'log2', 'n_estimators': 30}\n",
      "accuracy : 0.820967126193001\n"
     ]
    }
   ],
   "source": [
    "#grid search\n",
    "rfc=RandomForestClassifier(random_state=42)\n",
    "param_grid = { \n",
    "    'n_estimators': [10,20,30,40],\n",
    "    'max_features': ['auto', 'sqrt', 'log2'],\n",
    "    'max_depth' : [4,5,6,7,8,10,20,100,200,300],\n",
    "    'criterion' :['gini', 'entropy']\n",
    "}\n",
    "CV_rfc = GridSearchCV(estimator=rfc, param_grid=param_grid, cv= 5)\n",
    "CV_rfc.fit(X_train, y_train)\n",
    "CV_rfc.best_params_\n",
    "print(\"best parameters \",CV_rfc.best_params_)\n",
    "print(\"accuracy :\",CV_rfc.best_score_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
